core/domain/email_manager.py

```diff
@@ -164,5 +178,71 @@ def send_post_signup_email(user_id):
         EMAIL_FOOTER.value)
 
     _send_email(
-        user_id, feconf.SYSTEM_COMMITTER_ID, email_models.INTENT_SIGNUP,
+        user_id, feconf.SYSTEM_COMMITTER_ID, feconf.EMAIL_INTENT_SIGNUP,
         email_subject, email_body)
+
+
+def require_valid_intent(intent):
+    if intent not in feconf.VALID_MODERATOR_ACTIONS:
+        raise Exception('Unrecognized email intent: %s' % intent)
+
+
+def _get_email_config(intent):
+    require_valid_intent(intent)
+    return config_domain.Registry.get_config_property(
+        feconf.VALID_MODERATOR_ACTIONS[intent]['email_config'])
+
+
+def get_draft_moderator_action_email(intent):
+    """Returns a draft of the text of the body for an email sent immediately
+    following a moderator action.
+
+    If feconf.REQUIRE_EMAIL_ON_MODERATOR_ACTION is set to False, returns None.
+    """
+    if not feconf.REQUIRE_EMAIL_ON_MODERATOR_ACTION:
```

Comment:
You've already checked this in controllers. I think this file is justified in assuming that everything is legitimate and it can just send the emails.